### PR TITLE
Hello-containers training module

### PR DESCRIPTION
**Pull request by kenibrewer**
This PR adds a `hello-containers` training module that acts as a bridge to `hello-genomics` that teaches the basics of working with containers with and without Nextflow. 

Changes:
- 🧑‍💻 devxp(devcontainer): add pre-commit hooks, remove deprecated linters
- ✨ feat: add hello-containers training module
- Apply suggestions from code review
- 📝 docs: improve instructions for hello-containers

### CODE REVIEW

File: docs/hello_nextflow/03_hello_containers.md
Comment (mribeirodantas) [inline docs/hello_nextflow/03_hello_containers.md:265]:
```suggestion
_Before_:_
```

File: docs/hello_nextflow/03_hello_containers.md
Comment (mribeirodantas) [inline docs/hello_nextflow/03_hello_containers.md:243]:
```suggestion
!!!Hint: Even if Seqera Containers doesn't manage to successfully build a container image for you, the `Dockerfile` and `conda.yml` are great starting point for a manual build.
```

File: docs/hello_nextflow/03_hello_containers.md
Comment (mribeirodantas) [inline docs/hello_nextflow/03_hello_containers.md:290]:
```suggestion
!!!NOTE

```

**Review by mribeirodantas — CHANGES_REQUESTED** [review]
Can't run locally for some reason, but these changes should fix the issues I found on the preview.

File: hello-nextflow/hello-containers-4.2.nf
Comment (vdauwera) [inline hello-nextflow/hello-containers-4.2.nf:52]: add a comment?

**Review by vdauwera — CHANGES_REQUESTED** [review]
Content looks great, 1-3 do a fantastic job of bridging the gap between Hello World and GATK. Slight reservations about the 4th section since it gets a bit more advanced, would like to see a strong disclaimer about that section being more a spin-off, and make it clear that completing it is not a requirement in order to proceed to the next step in the training.

File: hello-nextflow/hello-containers.nf
Comment (kenibrewer) [reply hello-nextflow/hello-containers.nf:33]: Done

File: hello-nextflow/hello-containers-4.2.nf
Comment (kenibrewer) [reply hello-nextflow/hello-containers-4.2.nf:52]: Done

File: docs/hello_nextflow/03_hello_containers.md
Comment (vdauwera) [reply docs/hello_nextflow/03_hello_containers.md:164]: Or just lean harder into the whole "we're advertising seqera containers" thing and make the header be e.g. "Create a container via the Seqera Containers web service"

File: docs/hello_nextflow/03_hello_containers.md
Comment (vdauwera) [inline docs/hello_nextflow/03_hello_containers.md:172]: Needs to tell us why we're getting this package. Generally speaking, briefly explain the use case / goal before you dive into the specifics of how/what to do.

File: docs/hello_nextflow/03_hello_containers.md
Comment (vdauwera) [inline docs/hello_nextflow/03_hello_containers.md:205]: sentence doesn't quite work? Did you mean "You can see that"?

File: docs/hello_nextflow/03_hello_containers.md
Comment (vdauwera) [inline docs/hello_nextflow/03_hello_containers.md:245]: This sentence by itself is mildly concerning, it makes it sound like it might randomly work or not work. Maybe say more clearly that for cases that are not supported by Seqera, it's still possible to get some utility?

File: docs/hello_nextflow/03_hello_containers.md
Comment (vdauwera) [inline docs/hello_nextflow/03_hello_containers.md:164]: Even more generally though, this deeper review is making me realize that it feels like this section might be another slight distraction from the goal of this module, which is just to bridge the gap between Hello World and Hello Genomics (both of which use prebuilt containers that we hand to them). 

The key things we really need this module to cover are (1) running a container interactively so we can test commands, and (2) using a container from a workflow. I'd like to propose moving this current section 2 (get/build) to _after_ the section on running in a workflow. Then we can make it clear the first two sections are must-do for the training, and the latter two are take-home exercises or stretch goals, however we want to call them.

File: docs/hello_nextflow/03_hello_containers.md
Comment (vdauwera) [inline docs/hello_nextflow/03_hello_containers.md:259]: See comment earlier, propose we move this up to be section 2

File: docs/hello_nextflow/03_hello_containers.md
Comment (vdauwera) [inline docs/hello_nextflow/03_hello_containers.md:262]: make clear / reiterate this means you can run tools not installed in your compute env

File: docs/hello_nextflow/03_hello_containers.md
Comment (vdauwera) [inline docs/hello_nextflow/03_hello_containers.md:270]: add `linenums="<ln>"` whenever possible

File: docs/hello_nextflow/03_hello_containers.md
Comment (vdauwera) [inline docs/hello_nextflow/03_hello_containers.md:321]: check out (in two words)

File: docs/hello_nextflow/03_hello_containers.md
Comment (vdauwera) [inline docs/hello_nextflow/03_hello_containers.md:363]: I get to this bit and I'm like ahhhh that's why he wanted me to get the quote container. Reinforces that that bit belongs hereabouts, imo.

**Review by vdauwera — CHANGES_REQUESTED** [review]
Did a more thorough review, found a few minor quibbles; the biggest thing is I think the 3rd section should be pushed down to _after_ running from a workflow. 

Tbqh I'm still torn about 3 and 4 (especially the 4th section) because they're neat 'building on top of this stuff' add-ons but they don't really add anything that's super necessary wrt the main progression of the Hello series. They really are perfect spinoff module material. 

Also they're the two sections that need a bit more work to be ready for primetime so I'm tempted to put them on ice until after Summit. But if you're willing/able to find the bandwidth to sort them out, I'm open to keeping them in. WDYT?

File: docs/hello_nextflow/03_hello_containers.md
Comment (kenibrewer) [reply docs/hello_nextflow/03_hello_containers.md:9]: Added.

File: docs/hello_nextflow/03_hello_containers.md
Comment (kenibrewer) [reply docs/hello_nextflow/03_hello_containers.md:47]: I switched to a different container that doesn't have this requirement and forgot to update. Fixed

File: docs/hello_nextflow/03_hello_containers.md
Comment (kenibrewer) [reply docs/hello_nextflow/03_hello_containers.md:58]: tree isn't installed in many containers including this one. Sticking with `ls`

File: docs/hello_nextflow/03_hello_containers.md
Comment (kenibrewer) [reply docs/hello_nextflow/03_hello_containers.md:32]: No longer relevant based on changing container image.

File: docs/hello_nextflow/03_hello_containers.md
Comment (kenibrewer) [reply docs/hello_nextflow/03_hello_containers.md]: Added clarification about the goal of this section.

File: docs/hello_nextflow/03_hello_containers.md
Comment (kenibrewer) [reply docs/hello_nextflow/03_hello_containers.md]: Fixed.

File: docs/hello_nextflow/03_hello_containers.md
Comment (kenibrewer) [reply docs/hello_nextflow/03_hello_containers.md:285]: Done

File: docs/hello_nextflow/03_hello_containers.md
Comment (kenibrewer) [reply docs/hello_nextflow/03_hello_containers.md]: Agreed. Moved.

**Review by kenibrewer — COMMENTED** [review]
Made most of the edits requested. Waiting on re-organizing until after re-review.

**Comment by vdauwera** [comment]
Minor observations noted in Slack. Overall it’s really good, the flow works, it hits all the key points we need, and the extra bits are clearly labeled as such. Ok to merge once you've addressed the last batch and moved Section 2 up.

**Review by vdauwera — APPROVED** [review]

**Pull request merged**
